TRUNK‑5841| PatientProgram Domain – Switching from Hibernate mappings…

snowvirus opened the pull request:
… to annotations

<!--- Add a pull request title above in this format -->
<!--- real example: 'TRUNK-5111: Replace use of deprecated isVoided' -->
<!--- 'TRUNK-JiraIssueNumber: JiraIssueTitle' -->
## Description of what I changed
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->
i have changed PatientProgram to support annotation
i have tested the changes and all passed successful including `mvn clean install -Pskitp-default-test  -Pintegration-test`


## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first! -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it! -->
<!--- Just add the issue number at the end: -->
[see issue](https://openmrs.atlassian.net/browse/TRUNK-5841)

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

snowvirus commented:
@dkayiwa @ibacher @anyone 
is there anythimg `missing` to get this PR `merged` 
i have never got any

dkayiwa commented:
This is on dev3 which does not have your changes because they are not yet merged.
You can run your changes by going to the `webapp` folder and run `mvn jetty:run`

snowvirus commented:
@dkayiwa sorry for the first screen short. though i have failed to install the `demo data` i have added them manually. but i have added a program nor. hope i can be mergerd and other PRs if possible

<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/7173715e-e9ea-4ca0-a5f3-0a4e36f2733b" />

dkayiwa commented:
Are you able to add patients to those programs via the patient dashboard?

snowvirus commented:
> Are you able to add patients to those programs via the patient dashboard?

<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/93b566bb-82d3-40da-ab43-1249fee5dcf5" />

<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/920db1e8-e16c-4e11-a121-4dcb864402b8" />

@dkayiwa is that still not enough?

snowvirus commented on api/src/main/java/org/openmrs/PatientProgram.java:
```diff
@@ -37,22 +53,39 @@ public class PatientProgram extends BaseChangeableOpenmrsData implements Customi
 	private Date dateEnrolled;
 	
+	@Column(name = "date_completed")
+	@Temporal(TemporalType.TIMESTAMP)
```
@dkayiwa  `@Temporal` is used in JPA to define how a `java.util.Date` should be mapped to the database. Although we're using `java.util.Date` in the code, JPA doesn’t know by default whether we want to store just the date, just the time, or both.

Hope i haven't messed up!?

dkayiwa replied:
Is that what is being done for other fields of the same type for domains that are already converted to annotations?

snowvirus commented:
@dkayiwa @ibacher anything missing, so as I get merged?

dkayiwa commented:
@snowvirus are you still working on this?
